assembl/static2/js/app/components/voteSession/availableTokens.jsx

```diff
@@ -14,33 +15,98 @@ type Props = {
   tokenCategories: Array<?TokenCategory>
 };
 
-const AvailableTokens = ({ remainingTokensByCategory, sticky, tokenCategories }: Props) => (
-  <div className={classnames({ 'available-tokens-sticky': sticky, box: sticky, 'available-tokens': !sticky })}>
-    {tokenCategories.map((category, idx) => {
-      if (category) {
-        const { color, id, title, totalNumber } = category;
-        const remaining = remainingTokensByCategory.get(category.id);
-        return (
-          <div key={id}>
-            <div className="category-available-tokens">
-              <div className="text">
-                <h2 className="dark-title-2">{title}</h2>
-                <Translate value="debate.voteSession.remainingTokens" count={remaining} />
-              </div>
-              <div className="tokens">
-                {range(totalNumber).map(n => (
-                  <Circle key={n + 1} size="32px" strokeColor={color} fillColor={n + 1 <= remaining ? color : undefined} />
-                ))}
-              </div>
-            </div>
-            {idx !== tokenCategories.length - 1 && <div className="separator" />}
-          </div>
-        );
-      }
+const getColumnSizes: Function = (numberCategoriesToDisplay) => {
+  switch (numberCategoriesToDisplay) {
+  case 1:
+    return [12];
+  case 2:
+    return [6, 6];
+  case 3:
+    return [4, 4, 4];
+  case 4:
+    return [6, 6, 6, 6];
+  default:
+    return [12];
+  }
+};
+
+const getClassNames: Function = (numberCategoriesToDisplay) => {
+  switch (numberCategoriesToDisplay) {
+  case 1:
+    return ['left'];
+  case 2:
+    return ['left', 'right'];
+  case 3:
+    return ['left', '', 'right'];
+  case 4:
+    return ['left', 'right', 'left', 'right'];
+  default:
+    return [12];
```

Comment:
Huh? Returning an `Array<Int>` instead of `Array<string>`? I'm not a fan of functions that return different types. But, I understand you're trying to yield the default 12 grids on a large screen.